pymongo/collection.py

```diff
@@ -2280,6 +2279,178 @@ def index_information(
             info[index.pop("name")] = index
         return info
 
+    def list_search_indexes(
+        self,
+        name: Optional[str] = None,
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> CommandCursor[_DocumentType]:
```

Comment:
list_indexes avoids using the document_class/_DocumentType so I think we should avoid it here as well. The reason is _DocumentType is intended to represent documents in the collection, not an index model document. list_indexes uses SON because it predated ordered dictionaries but here dict is probably the best option.

Reply:
Fixed